Refresh.Database/GameDatabaseContext.Pins.cs

```diff
@@ -18,41 +18,39 @@ public void UpdateUserPinProgress(Dictionary<long, int> pinProgressUpdates, Game
         ];
 
-        this.Write(() => 
+        foreach (KeyValuePair<long, int> pinProgressUpdate in pinProgressUpdates)
         {
-            foreach (KeyValuePair<long, int> pinProgressUpdate in pinProgressUpdates)
-            {
-                long pinId = pinProgressUpdate.Key;
-                int newProgress = pinProgressUpdate.Value;
-                PinProgressRelation? existingProgress = existingProgresses.FirstOrDefault(p => p.PinId == pinId);
+            long pinId = pinProgressUpdate.Key;
+            int newProgress = pinProgressUpdate.Value;
+            PinProgressRelation? existingProgress = existingProgresses.FirstOrDefault(p => p.PinId == pinId);
 
-                if (existingProgress == null)
-                {
-                    PinProgressRelation newRelation = new()
-                    {
-                        PinId = pinId,
-                        Progress = newProgress,
-                        Publisher = user,
-                        FirstPublished = now,
-                        LastUpdated = now,
-                        IsBeta = isBeta,
-                    };
-                    this.PinProgressRelations.Add(newRelation);
-                    continue;
-                }
-
-                bool isSpecialTreatmentPin = descendingProgressPins.Contains(pinId);
-
-                // Only update progress if it's better. For most pins it's better the greater it is, but for the pins in
-                // specialTreatmentPins, it's better the smaller it is.
-                if (!isSpecialTreatmentPin && newProgress > existingProgress.Progress
-                  || isSpecialTreatmentPin && newProgress < existingProgress.Progress)
+            if (existingProgress == null)
+            {
+                PinProgressRelation newRelation = new()
                 {
-                    existingProgress.Progress = newProgress;
-                    existingProgress.LastUpdated = now;
-                }
+                    PinId = pinId,
+                    Progress = newProgress,
+                    Publisher = user,
+                    FirstPublished = now,
+                    LastUpdated = now,
+                    IsBeta = isBeta,
+                };
+                this.PinProgressRelations.Add(newRelation);
+                continue;
+            }
+
+            bool isSpecialTreatmentPin = descendingProgressPins.Contains(pinId);
+
+            // Only update progress if it's better. For most pins it's better the greater it is, but for the pins in
+            // specialTreatmentPins, it's better the smaller it is.
+            if ((!isSpecialTreatmentPin && newProgress > existingProgress.Progress)
+                || (isSpecialTreatmentPin && newProgress < existingProgress.Progress))
+            {
+                existingProgress.Progress = newProgress;
+                existingProgress.LastUpdated = now;
             }
-        });
+        }
+        SaveChanges();
```

Comment:
```suggestion
        this.SaveChanges();
```

and everywhere else it's used